src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -162,37 +162,56 @@ class GoogleExtensions(
   override def publishGroup(id: WorkbenchGroupName): Future[Unit] =
     googlePubSubDAO.publishMessages(googleServicesConfig.groupSyncTopic, Seq(id.toJson.compactPrint))
 
-  override def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] =
-    onGroupUpdateRecursive(groupIdentities, Seq.empty).unsafeToFuture()
-
-  private def onGroupUpdateRecursive(groupIdentities: Seq[WorkbenchGroupIdentity], visitedGroups: Seq[WorkbenchGroupIdentity]): IO[Unit] =
+  override def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] = {
     for {
-      idsAndSyncDates <- groupIdentities.toList.traverse { id =>
-        directoryDAO.getSynchronizedDate(id).map(dateOption => id -> dateOption)
-      }
-      // only sync groups that have already been synchronized
-      messagesForIdsWithSyncDates = idsAndSyncDates.collect {
-        case (gn: WorkbenchGroupName, Some(_)) => gn.toJson.compactPrint
-        case (rpn: FullyQualifiedPolicyId, Some(_)) => rpn.toJson.compactPrint
+      // only sync groups that have been synchronized in the past
+      previouslySyncedIds <- groupIdentities.toList.traverseFilter { id =>
+        directoryDAO.getSynchronizedDate(id).map(dateOption => dateOption.map(_ => id))
       }
-      _ <- IO.fromFuture(IO(googlePubSubDAO.publishMessages(googleServicesConfig.groupSyncTopic, messagesForIdsWithSyncDates)))
-      ancestorGroups <- groupIdentities.toList.traverse { id =>
-        directoryDAO.listAncestorGroups(id)
+
+      // get all the messages for the previously synced groups
+      messages <- previouslySyncedIds.traverse {
+          case groupName: WorkbenchGroupName => getGroupPublishMessages(groupName)
+          case accessPolicyId: FullyQualifiedPolicyId => IO.pure(List(accessPolicyId.toJson.compactPrint))
       }
-      managedGroupIds = (ancestorGroups.flatten ++ groupIdentities).filterNot(visitedGroups.contains).collect {
+
+      // publish all the messages
+      _ <- IO.fromFuture(IO(publishMessages(messages.flatten)))
+
+    } yield ()
+  }.unsafeToFuture()
+
+  private def getGroupPublishMessages(groupName: WorkbenchGroupName): IO[List[String]] = {
```

Comment:
Nit - maybe rename this to `makeGroupPublishMessages` or `makeMessagesForGroupSync` or `constructMessagesForGroupSync` or something like that.  Until I read the method, I thought you were getting (retrieving) messages from the pub/sub queue or something.